R/dataframe.R

```diff
@@ -68,32 +68,21 @@ data_frame_ <- tibble_
 #' @export
 #' @rdname tibble
 lst <- function(...) {
-  lst_(lazyeval::lazy_dots(...))
-}
+  xs <- tidy_dots(..., .named = 500L)
 
-#' @export
-#' @rdname tibble
-lst_ <- function(xs) {
   n <- length(xs)
-  if (n == 0) return(list())
-
-  # If named not supplied, used deparsed expression
-  col_names <- names2(xs)
-  missing_names <- col_names == ""
-  if (any(missing_names)) {
-    deparse2 <- function(x) paste(deparse(x$expr, 500L), collapse = "")
-    defaults <- vapply(xs[missing_names], deparse2, character(1),
-      USE.NAMES = FALSE)
-    col_names[missing_names] <- defaults
+  if (n == 0) {
+    return(list())
   }
 
   # Evaluate each column in turn
-  output <- vector("list", n)
+  col_names <- names2(xs)
+  output <- new_lst(n)
   names(output) <- character(n)
 
   for (i in seq_len(n)) {
-    res <- lazyeval::lazy_eval(xs[[i]], output)
-    if (!is.null(res)) {
+    res <- rlang::tidy_eval(xs[[i]], discard_unnamed(output))
```

Comment:
Can we get rid of `rlang::`? Also, how/in which cases does the use of `discard_unnamed()` change the behavior of this function?